src/ConstructiveSolidGeometry/VolumePrimitives/Box.jl

```diff
@@ -71,12 +71,13 @@ function Box(CO, hX::TX, hY::TY, hZ::TZ, origin::PT, rotation::ROT) where {TX, T
 end
 
 #Unit convesion
-function Box(CO, hX::Quantity, hY::Quantity, hZ::Quantity, origin, rotation) 
-    _hX = to_internal_units(hX)
-    _hY = to_internal_units(hY)
-    _hZ = to_internal_units(hZ)
-    Box(CO, _hX, _hY, _hZ, origin, rotation)
-end
+Box(CO, hX::Quantity, hY, hZ, origin, rotation) = Box(CO, to_internal_units(hX), hY, hZ, origin, rotation)
+Box(CO, hX, hY::Quantity, hZ, origin, rotation) = Box(CO, hX, to_internal_units(hY), hZ, origin, rotation)
+Box(CO, hX, hY, hZ::Quantity, origin, rotation) = Box(CO, hX, hY, to_internal_units(hZ), origin, rotation)
+Box(CO, hX::Quantity, hY::Quantity, hZ, origin, rotation) = Box(CO, to_internal_units(hX), to_internal_units(hY), hZ, origin, rotation)
+Box(CO, hX, hY::Quantity, hZ::Quantity, origin, rotation) = Box(CO, hX, to_internal_units(hY), to_internal_units(hZ), origin, rotation)
+Box(CO, hX::Quantity, hY, hZ::Quantity, origin, rotation) = Box(CO, to_internal_units(hX), hY, to_internal_units(hZ), origin, rotation)
+Box(CO, hX::Quantity, hY::Quantity, hZ::Quantity, origin, rotation) = Box(CO, to_internal_units(hX), to_internal_units(hY), to_internal_units(hZ), origin, rotation)
```

Comment:
I am not a fan of these permutations...

What is the problem here and why doesn't this work without permutations?
`to_internal_units` also works for numbers without units and just returns the number:
https://github.com/JuliaPhysics/SolidStateDetectors.jl/blob/9dabbf63f698fb1c73b03f8b1fd7218fdf1a1e23/src/ConstructiveSolidGeometry/Units.jl#L8

This could be fixed by `RealQuantity` but I see that `RealQuantity` is only defined in SSD and not CSG and we are modifying CSG functions. Maybe it makes sense to define `RealQuantity` in CSG already and export it?

Then, the following might (?) work:
```julia
function Box(CO, hX::RealQuantity, hY::RealQuantity, hZ::RealQuantity, origin, rotation)
    _hX = to_internal_units(hX)
    _hY = to_internal_units(hY)
    _hZ = to_internal_units(hZ)
    Box(CO, _hX, _hY, _hZ, origin, rotation)
end

```